README.md

```diff
@@ -149,23 +150,62 @@ the creation of a new management pack project.
+
+### Template Project
+Every new project creates a file system that has the basic project structure required to develop and build and Management Pack.
+Each file and directory is discussed in depth in the [mp-init](doc/mp-init.md) documentation. `app/adapter.py` is the entry point
+for the your adapter, and the best starting point for a new user to begin their development journey. `adapter.py` is a template
+adapter that has comments throughout its code that allow users to replace the code in the template adapter and turn it into their own adapter. The methods inside the adapter template are all required and should not be deleted; however, the code inside the methods should be substituted to generate the desired adapter. Each method is used for a different function as described below:
+
+- test(adapter_instance):
+Performs a test connection using the information given to the adapter_instance to verify the adapter has been set up properly. A typical test connection will generally consist of:
+
+ 1. Read identifier values from adapter_instance that are required to connect to the target(s)
+ 2. Connect to the target(s), and retrieve some sample data
+ 3. Disconnect cleanly from the target (ensure this happens even if an error occurs)
+ 4. If any of the above failed, return an error, otherwise pass.
+
+- get_endpoints(adapter_instance):
+This method will be run before the 'test' method, and VMware Aria Operations will use
+the results to extract a certificate from each URL. If the certificate is not trusted by
+the VMware Aria Operations Trust Store, the user will be prompted to either accept or reject
+the certificate. If it is accepted, the certificate will be added to the AdapterInstance
+object that is passed to the 'test' and 'collect' methods. Any certificate that is
+encountered in those methods should then be validated against the certificate(s)
+in the AdapterInstance.
+
+- collect(adapter_instance):
+Performs a collection against the target host. A typical collection will generally consist of:
+1. Read identifier values from adapter_instance that are required to connect to the target(s)
+2. Connect to the target(s), and retrieve data
+3. Add the data into a CollectResult's objects, properties, metrics, etc
+4. Disconnect cleanly from the target (ensure this happens even if an error occurs)
+5. Return the CollectResult.
+
+- get_adapter_definition():
+Defines the object types and attribute types (metric/property) that are present
+in a collection. Setting these object types and attribute types helps VMware Aria Operations to
+validate, process, and display the data correctly.
+
+Each method represents a single request, and it can be tested individually using `mp-test`, which is covered in the following section. The adapter is stateless,therefore the test __test__ and __collect__ methods should be self contained. For further guidance on using the template project consult the [Walkthroughs](../README.md#walkthroughs) section.
```

Comment:
* I think this should be before the descriptions of the methods
* "self-contained" is probably not the right word, since they can certainly call other methods, libraries, files, etc. Also none of the other methods can store state either, not just the test and collect methods.

Maybe something like this:
```suggestion
Each method represents a single request, and it can be tested individually using `mp-test`, which is covered in the following section. The adapter is stateless, therefore the methods are not able to store any data for use in later method calls. For further guidance on using the template project consult the [Walkthroughs](../README.md#walkthroughs) section.
```